### DISCUSSION TITLE
APPLICATION_URL does nothing?

### DISCUSSION THREAD

**Discussion by keithzg** [leepeuker/movary#621] [Support]
I have movary installed to a subfolder on the domain/webserver I'm running it on, but it's all broken, every link on the homepage at  `https://my.domain/movary` just goes to `https://my.domain//login.php` or such rather than respecting the `APPLICATION_URL` setting in `.env`.

**Comment by leepeuker** [comment]
how are you hosting your instance? docker? if yes how are you setting the env variable `APPLICATION_URL`

**Reply by keithzg** [reply]
Not docker, just directly on my webserver. I'm setting `APPLICATION_URL` in the `.env` file with the rest of the settings.

**Reply by keithzg** [reply]
More specifically, I followed the instructions at https://docs.movary.org/install/manual/ to install, and the `.env` file I'm editing is the one created by copying the `.env.production.example` in step 3. I've been assuming (especially since most of the options documented are in the example file) that this is configured according to https://docs.movary.org/configuration/. My config looks basically like this:

```ini
# Environment
ENV=production
TIMEZONE="America/Edmonton"
MIN_RUNTIME_IN_SECONDS_FOR_JOB_PROCESSING=15
APPLICATION_URL="https://example.com/movary/"

# Database
DATABASE_MODE=mysql
DATABASE_MYSQL_HOST=localhost
DATABASE_MYSQL_PORT=
DATABASE_MYSQL_NAME=movary
DATABASE_MYSQL_USER=SomeUserICreated
DATABASE_MYSQL_PASSWORD="And a password for them"
DATABASE_MYSQL_CHARSET=utf8mb4
```

I've tried both setting an explicit site with an apache2 config for it, and just putting a symlink to `/var/www/movary/public` into `/var/www/html` (which is being served by default at `/` by the webserver). In both instances, the main page doesn't seem to load right (missing images or styling) and the links it provides all use a URL missing the subfolder path. Weirder, while the "Movary" home link in the top corner goes to `https://example.com` (or rather my actual, super-secret address), all the other links have a double slash after their initial folder portion of the URL, ex. `https://example.com/users//actors`.

Thinking that when you asked how I was setting the variable that might mean there's more ways than one to skin this, I tried setting it in the site config too, but that didn't change anything. Here's my entire `/etc/apache2/sites-enabled/movary.conf` in case *that* shows something I'm doing that's silly and wrong or otherwise illuminates things:

```xml
Alias /movary "/var/www/movary/public"

<Directory /var/www/movary/public/>
  Options +FollowSymlinks
  AllowOverride All

  SetEnv APPLICATION_URL https://kloud.keithzg.ca/movary

</Directory>
```

**Reply by keithzg** [reply]
I assume it's reading the `.env` file just fine generally, because after performing the undocumented step of running `php bin/console.php database:migration:migrate` inbetween the https://docs.movary.org/install/manual/ and https://docs.movary.org/getting-started/ steps I see all the tables created in the MariaDB database I specified, and the running the user creation as per the getting-started doc page I see my new user if I run `select * from movary.user;`. So at very least it's not outright ignoring the file.

If I manually go to what the links imply should be the right relative addresses, I just get 404 errors from Apache.

**Reply by leepeuker** [reply]
Hm okay thanks for the detailed information. I will look into it later and see if I can find what is causing your issue.

**Comment by leepeuker** [comment]
So it seems Movary currently does not fully support the `APPLICATION_URL` (it is only used at a few selected places). I have created a issue to fix this: https://github.com/leepeuker/movary/issues/622

**Comment by leepeuker** [comment]
This should now be fixed with [0.66.1](https://github.com/leepeuker/movary/releases/tag/0.66.1). Feedback appreciated!
